contracts/strategies/StrategyUSDT3pool.sol

```diff
@@ -0,0 +1,431 @@
+    }
+    
+    function deposit() public isAuthorized {
+        rebalance();
+        uint _want = (IERC20(want).balanceOf(address(this))).sub(tank);
+        if (_want > 0) {
+            IERC20(want).safeApprove(_3pool, 0);
+            IERC20(want).safeApprove(_3pool, _want);
+            uint v = _want.mul(1e30).div(ICurveFi(_3pool).get_virtual_price());
+            ICurveFi(_3pool).add_liquidity([0, 0, _want], v.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+        uint _bal = IERC20(_3crv).balanceOf(address(this));
+        if (_bal > 0) {
+            IERC20(_3crv).safeApprove(y3crv, 0);
+            IERC20(_3crv).safeApprove(y3crv, _bal);
+            yvERC20(y3crv).deposit(_bal);
+        }
+    }
+    
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(_3crv != address(_asset), "3crv");
+        require(y3crv != address(_asset), "y3crv");
+        balance = _asset.balanceOf(address(this));
+        _asset.safeTransfer(controller, balance);
+    }
+    
+    // Withdraw partial funds, normally used with a vault withdrawal
+    function withdraw(uint _amount) external {
+        require(msg.sender == controller, "!controller");
+
+        rebalance();
+        uint _balance = IERC20(want).balanceOf(address(this));
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+            tank = 0;
+        }
+        else {
+            if (tank >= _amount) tank = tank.sub(_amount);
+            else tank = 0;
+        }
+
+        address _vault = Controller(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        uint _fee = _amount.mul(withdrawalFee).div(DENOMINATOR);
+        IERC20(want).safeTransfer(Controller(controller).rewards(), _fee);
+        IERC20(want).safeTransfer(_vault, _amount.sub(_fee));
+    }
+
+    function _withdrawSome(uint _amount) internal returns (uint) {
+        uint _amnt = _amount.mul(1e30).div(ICurveFi(_3pool).get_virtual_price());
+        uint _amt = _amnt.mul(1e18).div(yvERC20(y3crv).getPricePerFullShare());
+        uint _before = IERC20(_3crv).balanceOf(address(this));
+        yvERC20(y3crv).withdraw(_amt);
+        uint _after = IERC20(_3crv).balanceOf(address(this));
+        return _withdrawOne(_after.sub(_before));
+    }
+
+    function _withdrawOne(uint _amnt) internal returns (uint) {
+        uint _before = IERC20(want).balanceOf(address(this));
+        IERC20(_3crv).safeApprove(_3pool, 0);
+        IERC20(_3crv).safeApprove(_3pool, _amnt);
+        ICurveFi(_3pool).remove_liquidity_one_coin(_amnt, 2, _amnt.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR).div(1e12));
```

Comment:
Why the additional div(1e12) compared to TUSD strat? Assuming is because of the weird USDT decimals?

Reply:
Right. y3crv is 18, 3crv is 18, and USDT is 6. so every time when switching between USDT to/3crv should be taken care of the unit.